### PR TITLE
Add access ctrls

### PR DESCRIPTION
**Pull request by TJZenkai**

### CODE REVIEW

**Comment by ejesse** [comment]
Looks good to me, just a question: do we care about being explicit about the "danger" messages in the tests?

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -7,5 +7,6 @@ class ApplicationController < ActionController::Base
   def first_time_visit
     cookies.permanent[:first_visit] = 1
     @first_visit = true
```
Comment (davidanugo) [inline app/controllers/application_controller.rb]: This line is no longer necessary.

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -40,7 +40,11 @@ def show; end
 
   # GET /products/new
   def new
-    @product = Product.new
+    if CONFIGS[:enable_asset_creation]
+      @product = Product.new
+    else
+      redirect_to '/', notice: { type: 'danger', message: 'Cannot access URL directly.' }
```
Comment (vigneshjb) [inline app/controllers/products_controller.rb]: Can we move the message to the config ? It is something that people might want to override to provide custom re-direct URLs.

**Comment by vigneshjb** [comment]
This looks great !! :) We can merge it once you are done with David's and my comment.

**Pull request merged**